updateHostsFile.py

```diff
@@ -1296,21 +1310,21 @@ def remove_old_hosts_file(old_file_path, backup):
         Whether or not to backup the existing hosts file.
     """
 
-    # Create if already removed, so remove won't raise an error.
-    open(old_file_path, "a").close()
+    # only go through possible backup and removal if the file exists
+    if os.path.isfile(old_file_path):
```

Comment:
let's simplify to:
```
if os.path.isfile(old_file_path) and backup:
```

Reply:
That changes the logic, because if you look at the code closely, so see the deletion always happens if the file exists, but backup is only done if the backup option is given.  So if you collapse the conditional, it ends up not working as expected.